### PR TITLE
Update LCD display instructions

### PR DESCRIPTION
**Pull request by JamesH65**
Needed updating for the FKMS change on Pi4.

### CODE REVIEW

File: hardware/display/README.md [hunk 1]
```diff
@@ -14,6 +14,26 @@ The DSI display is designed to work with Raspberry Pi models 2, 3 and 4 that hav
 
 LCD displays have an optimum viewing angle, and depending on how the screen is mounted it may be necessary to change the orientation of the display to give the best results. By default, the Raspberry Pi Touch Display and Raspberry Pi are set up to work best when viewed from slightly above, for example on a desktop. If viewing from below, you can physically rotate the display, and then tell the system software to compensate by running the screen upside down.
 
+### FKMS Mode
+
+FKMS mode is used by default on the Raspberry Pi 4B. FKMS uses the DRM/MESA libraries to provide graphics and 3D acceleration. 
+
+To set screen orientation when running the graphics desktop, select the `Screen Configuration` option from the `Preferences` menu. Right Click on the DSI display rectangle in the layout editor, select Orientation then the required option.
```
Comment (lurch) [inline hardware/display/README.md]: 'graphics desktop' -> 'graphical desktop' ?

Comment (JamesH65) [reply hardware/display/README.md]: exactly what I meant to type....thank [deity] for copy editors!

File: hardware/display/README.md [hunk 2]
```diff
@@ -14,6 +14,26 @@ The DSI display is designed to work with Raspberry Pi models 2, 3 and 4 that hav
...
+
+To set screen orientation when in console mode, you will need to edit the kernel command line to pass the required orientation to the system. 
+```bash
+sudo nano /boot/cmdline
```
Comment (lurch) [inline hardware/display/README.md]: Should this be `/boot/cmdline.txt` ?

File: hardware/display/README.md [hunk 3]
```diff
@@ -14,6 +14,26 @@ The DSI display is designed to work with Raspberry Pi models 2, 3 and 4 that hav
...
+```
+To rotate by 90, add the following to the cmdline, making sure everything is on the same line, do not add any carriage returns. Possible rotation values are 0,90,180 and 270.
```
Comment (lurch) [inline hardware/display/README.md]: Should probably be spaces between the commas?

File: hardware/display/README.md (hunk 2 above)
Comment (JamesH65) [reply hardware/display/README.md]: Did I mention about the need for copy editors!!!!

File: hardware/display/README.md [hunk 4]
```diff
@@ -14,6 +14,26 @@ The DSI display is designed to work with Raspberry Pi models 2, 3 and 4 that hav
...
+```
+video=DSI-1:800x480@60,rotate=90
+```
+NOTE:  In console mode is not possible to rotate the DSI display separately to the HDMI display, so if you have both attached they must both be set to the same value.
```
Comment (lurch) [inline hardware/display/README.md]: What happens if they're **not** set to the same rotation value? Will neither display work, or will only one of the displays work?

File: hardware/display/README.md [hunk 5]
```diff
@@ -14,6 +14,26 @@ The DSI display is designed to work with Raspberry Pi models 2, 3 and 4 that hav
...
+
+### Legacy Graphics Mode
+
+Legacy graphics mode is used by default on all Raspberry Pi Models prior to the Raspberry Pi 4B, and can also be used on the Raspberry Pi 4B if required, by disabling FKMS mode by commenting out the FKMS line in `config.txt`.
+
 To flip the display, add, anywhere in the file `\boot\config.txt`, the following line:
```
Comment (lurch) [inline hardware/display/README.md]: back-slashes here should be forward-slashes

**Comment by lurch** [comment]
Just noticed that this page (outside of the text being edited in this PR) mentions both `display_lcd_rotate` and `display_hdmi_rotate` - is one of them wrong?

File: hardware/display/README.md (hunk 5 above)
Comment (JamesH65) [reply hardware/display/README.md]: oooeerr. Will fix that up.

File: hardware/display/README.md (hunk 3 above)
Comment (lurch) [reply hardware/display/README.md]: And perhaps "rotate by 90" should be expanded to "rotate 90 degrees clockwise" ?

**Comment by lurch** [comment]
Does the FKMS mode support the "flip" option that the Legacy mode supports?

**Comment by lurch** [comment]
"The DSI display is designed to work with Raspberry Pi models 2, 3 and 4 that have mounting holes in a HAT footprint. Model 1 A/B boards are supported, but require additional mounting hardware to fit the HAT-dimensioned stand-offs on the display PCB."

Minor nitpick - which category do the Model 1 A+ and B+ fall into?

**Comment by 6by9** [comment]
> Does the FKMS mode support the "flip" option that the Legacy mode supports?

Yes, you can add ```reflect_x``` and ```reflect_y``` as well as ```rotate=N``` to the video= entry in cmdline.txt

**Comment by JamesH65** [comment]
> > Does the FKMS mode support the "flip" option that the Legacy mode supports?
> 
> Yes, you can add `reflect_x` and `reflect_y` as well as `rotate=N` to the video= entry in cmdline.txt

Does this need to be added to the PR?

**Comment by JamesH65** [comment]
Unless people want the reflect items added, I'm inclined to merge this. Any objections or improvements?

File: hardware/display/README.md [hunk 6]
```diff
@@ -14,7 +14,27 @@ The DSI display is designed to work with Raspberry Pi models 2, 3 and 4 that hav
...
+```
+video=DSI-1:800x480@60,rotate=90
+```
+NOTE:  In console mode is not possible to rotate the DSI display separately to the HDMI display, so if you have both attached they must both be set to the same value. If different the results are unspecified. 
+
+### Legacy Graphics Mode
+
+Legacy graphics mode is used by default on all Raspberry Pi Models prior to the Raspberry Pi 4B, and can also be used on the Raspberry Pi 4B if required, by disabling FKMS mode by commenting out the FKMS line in `config.txt`.
+
+To flip the display, add, anywhere in the file `/boot/config.txt`, the following line:
```
Comment (lurch) [inline hardware/display/README.md]: Maybe it's just me, but I think so many commas in one sentence looks a bit clunky. Would "To flip the display, add the following line to `config.txt` :" work?

File: hardware/display/README.md [hunk 7]
```diff
@@ -14,7 +14,27 @@ The DSI display is designed to work with Raspberry Pi models 2, 3 and 4 that hav
...
+
+FKMS mode is used by default on the Raspberry Pi 4B. FKMS uses the DRM/MESA libraries to provide graphics and 3D acceleration. 
+
+To set screen orientation when running the graphical desktop, select the `Screen Configuration` option from the `Preferences` menu. Right click on the DSI display rectangle in the layout editor, select Orientation then the required option.
+
+To set screen orientation when in console mode, you will need to edit the kernel command line to pass the required orientation to the system. 
+```bash
+sudo nano /boot/cmdline.txt
+```
+To rotate by 90, add the following to the cmdline, making sure everything is on the same line, do not add any carriage returns. Possible rotation values are 0, 90, 180 and 270.
```
Comment (lurch) [inline hardware/display/README.md]: The text further down the page says "rotate 90 degrees clockwise" - maybe that's better than just "rotate by 90" ?

File: hardware/display/README.md [hunk 8]
```diff
@@ -14,7 +14,27 @@ The DSI display is designed to work with Raspberry Pi models 2, 3 and 4 that hav
...
+```
+video=DSI-1:800x480@60,rotate=90
+```
+NOTE:  In console mode is not possible to rotate the DSI display separately to the HDMI display, so if you have both attached they must both be set to the same value. If different the results are unspecified. 
+
+### Legacy Graphics Mode
+
+Legacy graphics mode is used by default on all Raspberry Pi Models prior to the Raspberry Pi 4B, and can also be used on the Raspberry Pi 4B if required, by disabling FKMS mode by commenting out the FKMS line in `config.txt`.
```
Comment (lurch) [inline hardware/display/README.md]: Does using legacy graphics mode on the Pi 4B come with some limitations? Should they be mentioned here? :man_shrugging:

**Comment by andrum99** [comment]
> "The DSI display is designed to work with Raspberry Pi models 2, 3 and 4 that have mounting holes in a HAT footprint. Model 1 A/B boards are supported, but require additional mounting hardware to fit the HAT-dimensioned stand-offs on the display PCB."
> 
> Minor nitpick - which category do the Model 1 A+ and B+ fall into?

The Pi 1 model A+ and B+ are the same shape as the later boards, and have the mounting holes in the same place. Perhaps this sentence should instead read:

"The DSI display is designed to work on all models of Raspberry Pi, however early models (The Raspberry Pi 1 model A and B) will require additional mounting hardware to fit the HAT dimensioned stand-offs on the display PCB."

**Comment by JamesH65** [comment]
Have made a change similar to that suggested by @andrum99 , any other comment prior to merging?

File: hardware/display/README.md [hunk 9]
```diff
@@ -6,15 +6,35 @@ The Raspberry Pi Touch Display is an LCD display which connects to the Raspberry
 
 ## Board support
 
-The DSI display is designed to work with Raspberry Pi models 2, 3 and 4 that have mounting holes in a HAT footprint. Model 1 A/B boards are supported, but require additional mounting hardware to fit the HAT-dimensioned stand-offs on the display PCB.
+The DSI display is designed to work on all models of Raspberry Pi, however early models that do not have mounting holes (The Raspberry Pi 1 model A and B) will require additional mounting hardware to fit the HAT-dimensioned stand-offs on the display PCB.
```
Comment (lurch) [inline hardware/display/README.md]: "on all models" -> "with all models" ?
Lowercase 'T' on "The Raspberry Pi"

Sorry @JamesH65 :laughing: 

Wasn't there also something to do with the I2C / DSI connections being different on pre-40-pin-GPIO models of Raspberry Pi, or am I confusing that with something else?

Comment (JamesH65) [reply hardware/display/README.md]: You need to connect to the I2C on the GPIO on those earlier models. Its covered on the legacy page.

**Comment by andrum99** [comment]
LGTM

File: hardware/display/README.md [hunk 10]
```diff
@@ -6,15 +6,35 @@ The Raspberry Pi Touch Display is an LCD display which connects to the Raspberry
...
+
+To set screen orientation when in console mode, you will need to edit the kernel command line to pass the required orientation to the system. 
+```bash
+sudo nano /boot/cmdline.txt
+```
+To rotate by 90 degrees clockwise, add the following to the cmdline, making sure everything is on the same line, do not add any carriage returns. Possible rotation values are 0, 90, 180 and 270.
+```
+video=DSI-1:800x480@60,rotate=90
+```
+NOTE:  In console mode is not possible to rotate the DSI display separately to the HDMI display, so if you have both attached they must both be set to the same value. If different the results are unspecified.
```
Comment (andrum99) [inline hardware/display/README.md]: Insert 'it' between 'mode' and 'is'

File: hardware/display/README.md [hunk 11]
```diff
@@ -6,15 +6,35 @@ The Raspberry Pi Touch Display is an LCD display which connects to the Raspberry
...
+```
+To rotate by 90 degrees clockwise, add the following to the cmdline, making sure everything is on the same line, do not add any carriage returns. Possible rotation values are 0, 90, 180 and 270.
+```
+video=DSI-1:800x480@60,rotate=90
+```
+NOTE:  In console mode is not possible to rotate the DSI display separately to the HDMI display, so if you have both attached they must both be set to the same value. If different the results are unspecified. 
+
+### Legacy Graphics Mode
+
+Legacy graphics mode is used by default on all Raspberry Pi Models prior to the Raspberry Pi 4B, and can also be used on the Raspberry Pi 4B if required, by disabling FKMS mode by commenting out the FKMS line in `config.txt`. Note, legacy mode on the Raspberry Pi 4B has no 3D acceleration so it should only be used if you have a specific reason for needing it.
```
Comment (andrum99) [inline hardware/display/README.md]: Models -> models.

Note, -> Note:

File: hardware/display/README.md [hunk 12]
```diff
@@ -6,15 +6,35 @@ The Raspberry Pi Touch Display is an LCD display which connects to the Raspberry
...
+
+To flip the display, add the following line to the file `/boot/config.txt`,:
```
Comment (andrum99) [inline hardware/display/README.md]: take out the comma after config.txt

File: hardware/display/README.md (hunk 9 above)
Comment (lurch) [reply hardware/display/README.md]: Ooops, I missed the link to that page :eyes:

**Pull request merged**